daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -84,32 +90,93 @@ class IdeLedgerClient(
     })
   }
 
-  override def queryContractId(
+  private def lookupContractInstance(
       parties: OneAnd[Set, Ref.Party],
-      templateId: Identifier,
       cid: ContractId,
-  )(implicit
-      ec: ExecutionContext,
-      mat: Materializer,
-  ): Future[Option[ScriptLedgerClient.ActiveContract]] = {
+  ): Option[Value.ContractInstance] = {
+
     ledger.lookupGlobalContract(
       view = ScenarioLedger.ParticipantView(Set(), Set(parties.toList: _*)),
       effectiveAt = ledger.currentTime,
       cid,
     ) match {
       case ScenarioLedger.LookupOk(
             _,
-            Versioned(_, Value.ContractInstance(_, arg, _)),
+            Versioned(_, contractInstance),
             stakeholders,
           ) if parties.any(stakeholders.contains(_)) =>
-        Future.successful(Some(ScriptLedgerClient.ActiveContract(templateId, cid, arg)))
+        Some(contractInstance)
       case _ =>
         // Note that contrary to `fetch` in a scenario, we do not
         // abort on any of the error cases. This makes sense if you
         // consider this a wrapper around the ACS endpoint where
         // we cannot differentiate between visibility errors
         // and the contract not being active.
-        Future.successful(None)
+        None
+    }
+  }
+
+  override def queryContractId(
+      parties: OneAnd[Set, Ref.Party],
+      templateId: Identifier,
+      cid: ContractId,
+  )(implicit
+      ec: ExecutionContext,
+      mat: Materializer,
+  ): Future[Option[ScriptLedgerClient.ActiveContract]] = {
+    Future.successful(
+      lookupContractInstance(parties, cid).map { case Value.ContractInstance(_, arg, _) =>
+        ScriptLedgerClient.ActiveContract(templateId, cid, arg)
+      }
+    )
+  }
+
+  override def queryViewContractId(
+      parties: OneAnd[Set, Ref.Party],
+      interfaceId: Identifier,
+      cid: ContractId,
+  )(implicit
+      ec: ExecutionContext,
+      mat: Materializer,
+  ): Future[Option[Value]] = {
+
+    lookupContractInstance(parties, cid) match {
+
+      case None => Future.successful(None)
+
+      case Some(Value.ContractInstance(templateId, arg, _)) =>
+        val valueTranslator = new ValueTranslator(
+          pkgInterface = compiledPackages.pkgInterface,
+          requireV1ContractIdSuffix = false,
+        )
+
+        valueTranslator.translateValue(TTyCon(templateId), arg) match {
+          case Left(_) =>
+            // TODO https://github.com/digital-asset/daml/issues/14830
+            Future.successful(None) // or error?
```

Comment:
I think this would be an "impossible" error, right? if so, probably better to actually fail than to successfully return `None`, which would give the impression of the contract not existing.

Reply:
I agree. I also think this is an _impossible_ case. Just wanted to check!
I will error/crash here then instead.

Reply:
and in one other place